enumerables.rb

```diff
@@ -0,0 +1,166 @@
+# rubocop:disable Metrics/ModuleLength
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    size.times do |x|
+      yield(self[x])
+    end
+    self
+  end
+
+  def my_each_with_index(value = 0)
+    return to_enum(:my_each_with_index) unless block_given?
+
+    index = value
+    size.times do |x|
+      yield(self[x], index)
+      index += 1
+    end
+  end
+
+  def my_select(arg = nil)
+    return to_enum(:my_select) unless block_given? || !arg.nil?
+
+    arr = []
+
+    if !arg.nil?
+      my_each do |x|
+        arr.push(x) if arg(x)
+      end
+    else
+      my_each do |x|
+        arr.push(x) if yield(x)
+      end
+    end
+    arr
+  end
+
+  # rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
+  def my_all?(arg = nil)
+    return false unless block_given? || arg.nil? == false || empty?
+
+    if block_given?
+      my_each do |x|
+        return false unless yield(x)
+      end
+    elsif arg.class == Regexp
+      my_each do |x|
+        return false unless x.match(arg)
+      end
+    elsif arg.class == Numeric
+      my_each do |x|
+        return false unless x.class == arg
+      end
+    else
+      my_each do |x|
+        return false unless arg == x
+      end
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    return true if !block_given? && arg.nil? && !(self.select { |x| x }).empty?
+    return false unless block_given? || !arg.nil?
+
+    if block_given?
+      my_each do |x|
+        return true if yield(x) == true
+      end
+    elsif arg.class == Regexp
+      my_each do |x|
+        return true if x.match(arg)
+      end
+    elsif (arg.is_a? Numeric) || (arg.is_a? String)
+      my_each do |x|
+        return true if arg == x
+      end
+    else
+      my_each do |x|
+        return true if x.class == arg
+      end
+      false
+    end
+  end
```

Comment:
# my_any?
1. [ ] Returns false if the block ever returns a value false or nil
### Test 👉 `array.my_any?(&false_block)`

|Expected|Got|
|:---:|:---:|
|`false`|` [8, 3, 7, 7, 3, 4, 7, 6, 2, 5, 0, 3, 6, 8, 2, 1, 8, 6, 3, 8, 3, 8, 3, 4, 2, 4, 4, 4, 5, 6, 7, 5, 2, 8..., 4, 4, 1, 6, 2, 5, 0, 4, 3, 4, 1, 1, 5, 3, 1, 0, 7, 6, 1, 3, 0, 7, 2, 0, 2, 2, 0, 2, 2, 0, 6, 7, 6]`| 
#
2. [ ] Returns false if the block ever returns a value false or nil:: range
### Test 👉 `range.my_any?(&false_block)`

Resulted in *NoMethodError*
#
3. [ ] When a class is passed as an argument returns true if at least one of the collection is a member of such class:: Numeric

|Expected|Got|
|:---:|:---:|
|`true`|`false`| 
#
4. [ ] When a Regex is passed as an argument returns false if none of the collection matches the Regex
### Test 👉 `words.my_any?(/z/)`

|Expected|Got|
|:---:|:---:|
|`false`|`#<Array:2080> => ["dog", "door", "rod", "blade"]`| 
#
5. When a pattern other than Regex or a Class is given returns false if none of the collection matches the pattern
### Test 👉 `words.my_any?('cat')`

|Expected|Got|
|:---:|:---:|
|`false`|`#<Array:2080> => ["dog", "door", "rod", "blade"]`|